numpy/core/shape_base.py

```diff
@@ -499,7 +499,16 @@ def _block(arrays, max_depth, result_ndim, depth=0):
         return _atleast_nd(arrays, result_ndim)
 
 
-# TODO: support array_function_dispatch
+def _block_dispatcher(arrays):
+    if type(arrays) is list:
```

Comment:
I'm not sure if we'll be breaking backward compatibility by doing this. Maybe `isinstance(arrays, collections.Iterable) and not hasattr(arrays, '__array_function__')`?

Reply:
My reasoning for picking `type(arrays) is list` is that it matches `np.block()`, which hard codes `list` (not even `tuple`). And we know that `list.__array_function__` will never exist.

Reply:
Okay, in that case ignore my comment. I was worried about the case where we might be inadvertently supporting other kinds of containers in the function but not the dispatcher. A comment would be nice, in any case.

Reply:
Yes, a comment is definitely a good idea here. Done!